botocore/awsrequest.py

```diff
@@ -337,6 +340,26 @@ def _urljoin(endpoint_url, url_path, host_prefix):
     return reconstructed
 
 
+def compress_aws_request(config, request, operation_model):
+    botocore.utils.validate_compression_config(config)
+    compressor = AWSRequestCompressor()
+    # ``AWSRequest.body`` is a property that will compute a URL encoded value
+    # if its a dictionary. So we shouild call it once hear to minimize recalculation.
+    body = request.body
+    if compressor.should_compress_request(
+        config=config,
+        operation_model=operation_model,
+        body=body,
+    ):
+        # AWSRequest.data is the raw input for the request body. This is the parameter
+        # that should be updated to properly construct an AWSPreparedRequest object.
+        request.data = compressor.compress(
+            body=body,
+            headers=request.headers,
+            encodings=operation_model.request_compression['encodings'],
+        )
```

Comment:
`compressor.compress` mixes in-place modification (of the `request.headers` dict) and returning a modified value (of `request.body`). Mixing the two styles is surprising by itself in Python, but for two properties of the same object?

Is there anything that can be done to make this behavior less surprising?

Reply:
Sure I can separate these into two separate methods or just modify the headers in the top-level function.

Reply:
Sure I can separate these into two separate methods or just modify the headers in the top-level function.